File: php/features.php
<?php
declare(strict_types=1);

// Dedline SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DedlineFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DedlineBaseFeature();
            case "test":
                return new DedlineTestFeature();
            default:
                return new DedlineBaseFeature();
        }
    }
}

File: py/features.py
# Dedline SDK feature factory

from feature.base_feature import DedlineBaseFeature
from feature.test_feature import DedlineTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DedlineBaseFeature(),
        "test": lambda: DedlineTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
